MobileBuy/buy/src/main/java/com/shopify/buy/dataprovider/ProductServiceDefault.java

```diff
@@ -239,13 +260,12 @@ public CancellableTask getProducts(final int page, final Long collectionId, Set<
             throw new IllegalArgumentException("page is a 1-based index, value cannot be less than 1");
         }
 
-        String sortOrderStr = sortOrder != null ? sortOrder.toString() : Collection.SortOrder.COLLECTION_DEFAULT.toString();
         if (collectionId == null) {
-            sortOrderStr = null;
+            throw new NullPointerException("collectionId cannot be null");
         }
 
+        final String sortOrderStr = sortOrder != null ? sortOrder.toString() : Collection.SortOrder.COLLECTION_DEFAULT.toString();
         final String tagsQueryStr = tags != null && !tags.isEmpty() ? TextUtils.join(",", tags.toArray()) : null;
```

Comment:
We do the same ternary operator above - how about breaking this out into a seperate function.